### PR TITLE
Add checkbox 'show as agenda item'. (Fixes #1783)

### LINKED ISSUE #1783: Set agenda item to hidden in create view of customslide/motion/assignment
In the create view we should provide an "item is hidden" checkbox to hide the created item directly in agenda. How can we get the corresponding agenda item after related object is created? 

Currently you have to open the QuickEdit-mode to set the "hidden" property of created agenda item.

### PR DESCRIPTION
**Pull request by emanuelschuetze**
Used for customslide, motion and assignment forms.
Changed default value of agenda.type to HIDDEN_ITEM (=2) in model.py.

@ostcar Please review carfully! I'm not sure if this the best way to do it.

### CODE REVIEW

File: openslides/assignments/static/js/assignments/site.js [hunk 1]
```diff
@@ -408,14 +427,23 @@ angular.module('OpenSlidesApp.assignments.site', ['OpenSlidesApp.assignments'])
...
         $scope.alert = {};
         // set initial values for form model by create deep copy of assignment object
         // so list/detail view is not updated while editing
         $scope.model = angular.copy(assignment);
         // get all form fields
         $scope.formFields = AssignmentForm.getFormFields();
+        for (var i = 0; i < $scope.formFields.length; i++) {
+            if ($scope.formFields[i].key == "showAsAgendaItem") {
+                // get state from agenda item (hidden/internal or agenda item)
+                // TODO: fix error after reload ("assignment.agenda_item is undefined")
```
Comment (emanuelschuetze) [inline openslides/assignments/static/js/assignments/site.js]: @ostcar How can I fix this error? I already add `Assignment.loadRelations(assignment, 'agenda_item');`... If I go to agenda and then back to AssignmentForm it works.

**Comment by emanuelschuetze** [comment]
Error resolved. Rebased. Ready for review.

File: openslides/assignments/static/js/assignments/site.js [hunk 2]
```diff
@@ -396,6 +415,22 @@ angular.module('OpenSlidesApp.assignments.site', ['OpenSlidesApp.assignments'])
         $scope.save = function(assignment) {
             Assignment.create(assignment).then(
                 function(success) {
+                    // find related agenda item
+                    Agenda.find(success.agenda_item_id).then(function(item) {
```
Comment (ostcar) [inline openslides/assignments/static/js/assignments/site.js]: You can use loadRelations for this

File: openslides/assignments/static/js/assignments/site.js [hunk 3]
```diff
@@ -88,13 +92,18 @@ angular.module('OpenSlidesApp.assignments.site', ['OpenSlidesApp.assignments'])
...
+    'operator',
+    function (gettextCatalog, operator) {
         return {
             // ngDialog for assignment form
             getDialog: function (assignment) {
                 if (assignment) {
                     var resolve = {
-                        assignment: function(Assignment) {return Assignment.find(assignment.id);}
+                        assignment: function(Assignment) {
+                            return Assignment.find(assignment.id).then(function(assignment) {
```
Comment (ostcar) [inline openslides/assignments/static/js/assignments/site.js]: What is "assignment"? If it is an assignment, then you don't have to "find" it

File: openslides/assignments/static/js/assignments/site.js [hunk 4]
```diff
@@ -396,6 +415,22 @@ angular.module('OpenSlidesApp.assignments.site', ['OpenSlidesApp.assignments'])
...
+                            // set item type to AGENDA_ITEM = 1
+                            type = 1;
+                        } else {
+                            // set item type to HIDDEN_ITEM = 2
+                            type = 2;
+                        }
+                        // save only if agenda item type is modified
+                        if (type != item.type) {
+                            item.type = type;
+                            Agenda.save(item);
```
Comment (ostcar) [inline openslides/assignments/static/js/assignments/site.js]: Proposal
```javascript
var type = assignment.showAsAgendaItem ? 1 : 2;
if (item.type != type) {
  item.type = type;
  Agenda.save(item)
}

File: openslides/assignments/static/js/assignments/site.js [hunk 5]
```diff
@@ -408,22 +443,43 @@ angular.module('OpenSlidesApp.assignments.site', ['OpenSlidesApp.assignments'])
...
+                        // set item type to AGENDA_ITEM = 1
+                        type = 1;
+                    } else {
+                        // set item type to HIDDEN_ITEM = 2
+                        type = 2;
+                    }
+                    // save only if agenda item type is modified
+                    if (type != assignment.agenda_item.type) {
+                        assignment.agenda_item.type = type;
+                        Agenda.save(assignment.agenda_item);
```
Comment (ostcar) [inline openslides/assignments/static/js/assignments/site.js]: See proposal above

File: openslides/motions/static/js/motions/site.js [hunk 6]
```diff
@@ -153,7 +157,11 @@ angular.module('OpenSlidesApp.motions.site', ['OpenSlidesApp.motions'])
             getDialog: function (motion) {
                 if (motion) {
                     var resolve = {
-                        motion: function(Motion) { return Motion.find(motion.id); }
+                        motion: function(Motion) {
+                            return Motion.find(motion.id).then(function(motion) {
```
Comment (ostcar) [inline openslides/motions/static/js/motions/site.js]: The same as above. What is motion?

File: openslides/motions/static/js/motions/site.js [hunk 7]
```diff
@@ -614,6 +632,22 @@ angular.module('OpenSlidesApp.motions.site', ['OpenSlidesApp.motions'])
...
+                            type = 1;
+                        } else {
+                            // set item type to HIDDEN_ITEM = 2
+                            type = 2;
+                        }
+                        // save only if agenda item type is modified
+                        if (type != item.type) {
+                            item.type = type;
+                            Agenda.save(item);
+                        }
```
Comment (ostcar) [inline openslides/motions/static/js/motions/site.js]: Same proposal as above

File: openslides/motions/static/js/motions/site.js [hunk 8]
```diff
@@ -686,6 +725,20 @@ angular.module('OpenSlidesApp.motions.site', ['OpenSlidesApp.motions'])
...
+                        // set item type to AGENDA_ITEM = 1
+                        type = 1;
+                    } else {
+                        // set item type to HIDDEN_ITEM = 2
+                        type = 2;
+                    }
+                    // save only if agenda item type is modified
+                    if (type != motion.agenda_item.type) {
+                        motion.agenda_item.type = type;
+                        Agenda.save(motion.agenda_item);
```
Comment (ostcar) [inline openslides/motions/static/js/motions/site.js]: Same proposal as above

File: openslides/assignments/static/js/assignments/site.js (hunk 2 above)
Comment (emanuelschuetze) [inline openslides/assignments/static/js/assignments/site.js]: I tested it with loadRelations but it does not work as expected.
```
Assignment.loadRelations(assignment, 'agenda_item').then(function() {
                        console.log(assignment.agenda_item);
```
 I have to wait until the promise is returns that I can change the loaded related agenda item. That works good with Agenda.find(). Please give me a code example if you have a better solution.

File: openslides/assignments/static/js/assignments/site.js (hunk 3 above)
Comment (emanuelschuetze) [inline openslides/assignments/static/js/assignments/site.js]: Fixed.

File: openslides/assignments/static/js/assignments/site.js (hunk 4 above)
Comment (emanuelschuetze) [inline openslides/assignments/static/js/assignments/site.js]: Done.

File: openslides/assignments/static/js/assignments/site.js (hunk 5 above)
Comment (emanuelschuetze) [inline openslides/assignments/static/js/assignments/site.js]: Done.

File: openslides/motions/static/js/motions/site.js (hunk 6 above)
Comment (emanuelschuetze) [inline openslides/motions/static/js/motions/site.js]: Fixed.

File: openslides/motions/static/js/motions/site.js (hunk 7 above)
Comment (emanuelschuetze) [inline openslides/motions/static/js/motions/site.js]: Done

File: openslides/motions/static/js/motions/site.js (hunk 8 above)
Comment (emanuelschuetze) [inline openslides/motions/static/js/motions/site.js]: Done.

**Comment by emanuelschuetze** [comment]
All done. Rebased. Ready for last review and merge.

**Comment by normanjaeckel** [comment]
Ok.

**Comment by emanuelschuetze** [comment]
@ostcar Ready for merge.

**Pull request merged**